packages/compass-explain-plan/src/components/explain-stage/clock.tsx

```diff
@@ -0,0 +1,303 @@
+        .endAngle(2 * Math.PI)
+    )
+    .attr('transform', `translate(${width / 2},${height / 2})`)
+    .style('fill', fillColor);
+
+  // border
+  d3.select(svgElement)
+    .append('path')
+    .attr(
+      'd',
+      d3.svg
+        .arc()
+        .innerRadius(radius - 1)
+        .outerRadius(radius)
+        .startAngle(0)
+        .endAngle(2 * Math.PI)
+    )
+    .attr('transform', `translate(${width / 2},${height / 2})`)
+    .style('fill', strokeColor);
+
+  // clock position lines
+  const angles = [0, 45, 90, 135, 180, 225, 270, 315];
+  const longLineLength = 0.25 * radius;
+  const shortLineLength = 0.16 * radius;
+
+  angles.forEach((angle) => {
+    const length = angle % 10 === 0 ? longLineLength : shortLineLength;
+    d3.select(svgElement)
+      .append('line')
+      .attr(
+        'x1',
+        radius + (radius - length) * Math.sin(angle * (Math.PI / 180))
+      )
+      .attr(
+        'y1',
+        radius - (radius - length) * Math.cos(angle * (Math.PI / 180))
+      )
+      .attr('x2', radius + radius * Math.sin(angle * (Math.PI / 180)))
+      .attr('y2', radius - radius * Math.cos(angle * (Math.PI / 180)))
+      .attr('stroke', strokeColor)
+      .attr('stroke-width', 2);
+  });
+}
+
+type ClockProps = {
+  totalExecTimeMS: number;
+  curStageExecTimeMS: number;
+  prevStageExecTimeMS: number;
+  width: number;
+  height: number;
+  className?: string;
+};
+
+const Clock: React.FunctionComponent<ClockProps> = ({
+  totalExecTimeMS,
+  curStageExecTimeMS,
+  prevStageExecTimeMS,
+  className,
+  width,
+  height,
+}) => {
+  const svgRef = useRef<SVGSVGElement>(null);
+
+  const deltaExecTime = useMemo(
+    () => curStageExecTimeMS - prevStageExecTimeMS,
+    [curStageExecTimeMS, prevStageExecTimeMS]
+  );
+
+  const darkmode = useDarkMode();
+
+  const {
+    clockFaceColor,
+    clockBackgroundColor,
+    textColor,
+    msColor,
+    previusElapsedArcColor,
+    currentElapsedArcColor,
+  } = useMemo(() => {
+    const lightModeColors = {
+      clockBackgroundColor: palette.white,
+      clockFaceColor: palette.gray.light1,
+      textColor: palette.gray.dark1,
+      msColor: palette.blue.light1,
+      previusElapsedArcColor: palette.gray.light1,
+      currentElapsedArcColor: palette.blue.light1,
+    };
+
+    const darkModeColors = {
+      clockBackgroundColor: palette.black,
+      clockFaceColor: palette.gray.dark1,
+      textColor: palette.gray.dark1,
+      msColor: palette.blue.light1,
+      previusElapsedArcColor: palette.gray.dark1,
+      currentElapsedArcColor: palette.blue.light1,
+    };
+
+    return darkmode ? darkModeColors : lightModeColors;
+  }, [darkmode]);
+
+  const {
+    containerStyles,
+    faceContainerStyle,
+    executionTimeStyles,
+    msecsStyles,
+    svgStyles,
+  } = useMemo(() => {
+    const containerStyles = css({
+      fontWeight: 'normal',
+      fontSize: '10px',
+      textAlign: 'center',
+      color: textColor,
+      height,
+      width,
+    });
```

Comment:
I would suggest moving static styles out and setting dynamic through `style` attribute